README.md

```diff
@@ -0,0 +1,150 @@
+
+* The further down a segment occurs on a page (the greater the segment index),
+  the less priority the segment receives. This is a minuscule change of priority.
+* Pages linked from main page get a multiplication factor of 5 to all segments.
+* Main page get a multiplication factor of 10 on all segments. 
+
+Basically this means the following order when synthesizing:
+1. All segements in the main page.
+2. The first segment in pages linked from the current main page.
+3. The second segment in pages linked from the current main page.
+4. ... until all segments in all pages links from the current main page is synthesized.
+5. The first segment in pages found in recent changes.
+6. The second segment in pages found in recent changes.
+7. ... until all segments in all pages found in recent changes has been synthesized.
+
+Candidates to be synthesized is re-evaluated every five minutes.
+
+### Automatic flushing of segments
+
+As the number of candidates to be synthesized can grow very large in a rather short time,
+a flushing mechanism kicks in when there are more than 100,000 candidates in the queue, 
+removing those with the lowest priority and retains the top 100,000.
+
+Segments flushing exists in order to save RAM, as the state of the application is kept in heap.
+
+### Automatic flushing of pages
+
+After one day of inactivity to a page on a wiki, 
+the state of rendering for that page will be a candidate for being flushed out.
+If there are still segments that have not been synthesized, this occurs after two days.
+
+Flushing a page means that if there is a change after the flush, 
+the complete page will be re-synthesized. 
+(Re-synthesized as in requested to be listened to. Wikispeech backend might in fact be cached.)
+
+The main page will never be flushed out.
+
+Pages that are linked to from the main page will not be flushed out until five days after they were last seen on the main page.
+
+Page flushing exists in order to save RAM, as the state of the application is kept in heap.
+  
+### Failing segment voices
+
+Will be a candidate to be retried every n hours, where n=number of previous failures.
+
+## TODO
+
+* Add feature in Wikispeech to not send audio response on synthesis, in order to minimize network data.
+* Add feature in Wikispeech to list all cached segments and voices for a given page, in order to avoid requesting synthesis when not needed.
+* Make all hard coded value mentioned above configurable in a properties file or something.
+* Report state of candidate, flushing, etc to influxdb.
+
+## REST
+
+Most REST calls are exist for debug and development reasons.
+As a user of this service, all you need is ```POST /api/wiki```
+
+### POST /api/wiki
+* consumerUrl: Wiki to be monitored for pre rendered.
+* initialLastRecentChangesLimitInMinutes: (default: 60) Number of hours of initial recent changes backlog to be processed.
+* mainPagePriority: (default: 10) Base priority of segments on Wiki main page.
```

Comment:
Should this be "Base priority **multiplier** of ..." per description under "## What this service does"